src/test/java/net/minestom/server/collision/EntityBlockTouchTickIntegrationTest.java

```diff
@@ -117,46 +202,53 @@ public void onTouch(@NotNull Touch touch) {
             }
         };
 
-        instance.setBlock(1000, 42, 1000, Block.STONE.withHandler(handler));
-        instance.setBlock(1000, 42, 1001, Block.STONE.withHandler(handler));
-        instance.setBlock(1000, 43, 1001, Block.STONE.withHandler(handler));
-        instance.setBlock(1000, 43, 999, Block.STONE.withHandler(handler));
-        instance.setBlock(1001, 43, 999, Block.STONE.withHandler(handler));
-        instance.setBlock(1001, 42, 999, Block.STONE.withHandler(handler));
-        instance.setBlock(1001, 42, 1001, Block.STONE.withHandler(handler));
-        instance.setBlock(1001, 43, 1000, Block.STONE.withHandler(handler));
-        instance.setBlock(999, 42, 1001, Block.STONE.withHandler(handler));
-        instance.setBlock(1001, 43, 1001, Block.STONE.withHandler(handler));
-        instance.setBlock(1001, 42, 1000, Block.STONE.withHandler(handler));
-        instance.setBlock(1000, 42, 1010, Block.STONE.withHandler(handler));
+        var customBlock = Block.STONE.withHandler(handler);
+        var offset = new Vec(1000, 0, 1000);
+        var blockPos = Set.of(offset.add(-1, 43, 0),
+                offset.add(1, 43, 0),
+                offset.add(0, 43, -1),
+                offset.add(0, 43, 1),
+                offset.add(0, 45, 0) // Y+
+        );
 
-        var entity = new Entity(EntityType.ZOMBIE);
-        entity.setInstance(instance, new Pos(1000.699, 42, 1000)).join();
+        for (var pos : blockPos) {
+            instance.setBlock(pos, customBlock);
+        }
+        instance.setBlock(1000, 42, 1000, Block.STONE); // Regular floor as we are going to be sliding into the blocks.
 
+        var entity = new Entity(EntityType.ZOMBIE);
+        var spawnPoint = new Pos(1000.5, 43, 1000.5);
+        entity.setInstance(instance, spawnPoint).join();
+
+        Arrays.stream(Direction.values())
+                .filter(direction -> direction != Direction.DOWN)
+                .map(Direction::vec)
+                .forEachOrdered(vec -> {
+                    entity.setVelocity(vec.mul(10));
+                    entity.tick(0);
+                    entity.teleport(spawnPoint);
+                });
+
+        assertEquals(blockPos, positions);
+
+        positions.clear(); // Final -Y check
+        instance.setBlock(1000, 42, 1000, customBlock);
         entity.tick(0);
-
-        assertEquals(Set.of(
-                new Vec(1000, 43, 999),
-                new Vec(1000, 42, 1000),
-                new Vec(1001, 43, 1000),
-                new Vec(1001, 42, 1000),
-                new Vec(1001, 42, 999),
-                new Vec(1001, 43, 999)
-            ), positions);
+        assertEquals(Set.of(new Vec(1000, 42, 1000)), positions);
 
         assertEquals(instance, entity.getInstance());
     }
 
     @Test
-    public void entityPhysicsCheckTouchTickFarNegative(Env env) {
+    public void entityPhysicsCheckTouchTickFarNegativeXZ(Env env) {
```

Comment:
Nitpick: Remove public for junit5.
Ref: [junit.org/junit5/docs/snapshot/user-guide#writing-tests-classes-and-methods](https://junit.org/junit5/docs/snapshot/user-guide/#writing-tests-classes-and-methods)